pep-9999.rst

```diff
@@ -0,0 +1,316 @@
+depends on the ``vsyscall`` page. As described in :pep:`571`, this
+breaks the assumption that a Docker containers userland is compatible
+wtih its host's kernel. Similar to ``manylinux2010``, Docker images
+for ``manylinux2014`` require patching ``glibc`` to remove all
+dependencies on ``vsyscall`` in the version of ``glibc`` included with
+our Docker image.
+
+Auditwheel
+----------
+
+The ``auditwheel`` tool has also been updated to produce
+``manylinux2014`` wheels. [9]_ Its behavior and purpose are otherwise
+unchanged from :pep:`513`.
+
+Platform Detection for Installers
+=================================
+
+Platforms may define a ``manylinux2014_compatible`` boolean attribute
+on the ``_manylinux`` module described in :pep:`513`. A platform is
+considered incompatible with ``manylinux2014`` if the attribute is
+``False``.
+
+If the ``_manylinux`` module is not found, or it does not have the
+attribute ``manylinux2014_compatible``, tools may fall back to
+checking for glibc. If the platform has glibc 2.17 or newer, it is
+assumed to be compatible unless the ``_manylinux`` module says
+otherwise.
+
+Specifically, the algorithm we propose is::
+
+    def is_manylinux2014_compatible():
+        # Only Linux, and only supported architectures
+        from distutils.util import get_platform
+
+        if get_platform() not in [
+            "linux-x86_64",
+            "linux-i686",
+            "linux-aarch64",
+            "linux-armhfp",
+            "linux-ppc64",
+            "linux-ppc64le",
+            "linux-s390x",
+        ]:
+            return False
+
+        # Check for presence of _manylinux module
+        try:
+            import _manylinux
+
+            return bool(_manylinux.manylinux2014_compatible)
+        except (ImportError, AttributeError):
+            # Fall through to heuristic check below
+            pass
+
+        # Check glibc version. CentOS 7 uses glibc 2.17.
+        # PEP 513 contains an implementation of this function.
+        return have_compatible_glibc(2, 17)
+
+Backwards compatibility with ``manylinux2010`` wheels
+=====================================================
+
+As explained in :pep:`513`, the specified symbol versions for
+``manylinux1`` allowed libraries constitute an *upper bound*.  The
+same is true for the symbol versions defined for ``manylinux2014`` in
+this PEP.  As a result, ``manylinux1`` and ``manylinux2010`` wheels
+are considered ``manylinux2014`` wheels.  A ``pip`` that recognizes
+the ``manylinux2014`` platform tag will thus install ``manylinux2010``
+wheels for ``manylinux2014`` platforms -- even when explicitly set --
+when no ``manylinux2014`` wheels are available.
+
+PyPI Support
+============
+
+PyPI should permit wheels containing the ``manylinux2014`` platform
+tag to be uploaded in the same way that it permits ``manylinux2010``.
+It should not attempt to verify the compatibility of ``manylinux2014``
+wheels.
```

Comment:
So if we resolve https://github.com/pypa/warehouse/issues/5420 will we need to make a new PEP to modify this provision?

Reply:
Yes, presumably!

Reply:
The PEP would need updating, but it can simply be a PEP modification, I would have thought.

Reply:
Either way, this is currently fine as-is.